//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, vJAXB 2.1.10 in JDK 6 
// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2010.08.09 at 01:59:40 PM PDT 
//


package gov.nih.nlm.cabio.jaxb.generated.pathways;

import java.util.ArrayList;
import java.util.List;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlType;


/**
 * <p>Java class for Gene complex type.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.
 * 
 * <pre>
 * &lt;complexType name="Gene">
 *   &lt;complexContent>
 *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *       &lt;sequence>
 *         &lt;element name="cytogeneticLocationCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneCytogeneticLocation" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="markerCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Marker" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="taxon" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Taxon" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="homologousAssociationCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}HomologousAssociation" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="exonArrayReporterCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}ExonArrayReporter" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="proteinCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Protein" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="geneOntologyCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneOntology" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="expressionArrayReporterCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}ExpressionArrayReporter" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="geneRelativeLocationCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneRelativeLocation" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="geneAliasCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneAlias" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="organOntologyCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}OrganOntology" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="geneFunctionAssociationCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneFunctionAssociation" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="nucleicAcidSequenceCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}NucleicAcidSequence" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="chromosome" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Chromosome" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="histopathologyCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Histopathology" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="transcriptCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Transcript" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="databaseCrossReferenceCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.common.domain}DatabaseCrossReference" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="physicalLocationCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GenePhysicalLocation" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="libraryCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Library" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="targetCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Target" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *         &lt;element name="pathwayCollection" minOccurs="0">
 *           &lt;complexType>
 *             &lt;complexContent>
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
 *                 &lt;sequence>
 *                   &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Pathway" maxOccurs="unbounded" minOccurs="0"/>
 *                 &lt;/sequence>
 *               &lt;/restriction>
 *             &lt;/complexContent>
 *           &lt;/complexType>
 *         &lt;/element>
 *       &lt;/sequence>
 *       &lt;attribute name="bigid" type="{http://www.w3.org/2001/XMLSchema}string" />
 *       &lt;attribute name="clusterId" type="{http://www.w3.org/2001/XMLSchema}long" />
 *       &lt;attribute name="fullName" type="{http://www.w3.org/2001/XMLSchema}string" />
 *       &lt;attribute name="hugoSymbol" type="{http://www.w3.org/2001/XMLSchema}string" />
 *       &lt;attribute name="id" type="{http://www.w3.org/2001/XMLSchema}long" />
 *       &lt;attribute name="symbol" type="{http://www.w3.org/2001/XMLSchema}string" />
 *     &lt;/restriction>
 *   &lt;/complexContent>
 * &lt;/complexType>
 * </pre>
 * 
 * 
 */
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "Gene", propOrder = {
    "cytogeneticLocationCollection",
    "markerCollection",
    "taxon",
    "homologousAssociationCollection",
    "exonArrayReporterCollection",
    "proteinCollection",
    "geneOntologyCollection",
    "expressionArrayReporterCollection",
    "geneRelativeLocationCollection",
    "geneAliasCollection",
    "organOntologyCollection",
    "geneFunctionAssociationCollection",
    "nucleicAcidSequenceCollection",
    "chromosome",
    "histopathologyCollection",
    "transcriptCollection",
    "databaseCrossReferenceCollection",
    "physicalLocationCollection",
    "libraryCollection",
    "targetCollection",
    "pathwayCollection"
})
public class Gene {

    protected Gene.CytogeneticLocationCollection cytogeneticLocationCollection;
    protected Gene.MarkerCollection markerCollection;
    protected Gene.Taxon taxon;
    protected Gene.HomologousAssociationCollection homologousAssociationCollection;
    protected Gene.ExonArrayReporterCollection exonArrayReporterCollection;
    protected Gene.ProteinCollection proteinCollection;
    protected Gene.GeneOntologyCollection geneOntologyCollection;
    protected Gene.ExpressionArrayReporterCollection expressionArrayReporterCollection;
    protected Gene.GeneRelativeLocationCollection geneRelativeLocationCollection;
    protected Gene.GeneAliasCollection geneAliasCollection;
    protected Gene.OrganOntologyCollection organOntologyCollection;
    protected Gene.GeneFunctionAssociationCollection geneFunctionAssociationCollection;
    protected Gene.NucleicAcidSequenceCollection nucleicAcidSequenceCollection;
    protected Gene.Chromosome chromosome;
    protected Gene.HistopathologyCollection histopathologyCollection;
    protected Gene.TranscriptCollection transcriptCollection;
    protected Gene.DatabaseCrossReferenceCollection databaseCrossReferenceCollection;
    protected Gene.PhysicalLocationCollection physicalLocationCollection;
    protected Gene.LibraryCollection libraryCollection;
    protected Gene.TargetCollection targetCollection;
    protected Gene.PathwayCollection pathwayCollection;
    @XmlAttribute
    protected String bigid;
    @XmlAttribute
    protected Long clusterId;
    @XmlAttribute
    protected String fullName;
    @XmlAttribute
    protected String hugoSymbol;
    @XmlAttribute
    protected Long id;
    @XmlAttribute
    protected String symbol;

    /**
     * Gets the value of the cytogeneticLocationCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.CytogeneticLocationCollection }
     *     
     */
    public Gene.CytogeneticLocationCollection getCytogeneticLocationCollection() {
        return cytogeneticLocationCollection;
    }

    /**
     * Sets the value of the cytogeneticLocationCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.CytogeneticLocationCollection }
     *     
     */
    public void setCytogeneticLocationCollection(Gene.CytogeneticLocationCollection value) {
        this.cytogeneticLocationCollection = value;
    }

    /**
     * Gets the value of the markerCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.MarkerCollection }
     *     
     */
    public Gene.MarkerCollection getMarkerCollection() {
        return markerCollection;
    }

    /**
     * Sets the value of the markerCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.MarkerCollection }
     *     
     */
    public void setMarkerCollection(Gene.MarkerCollection value) {
        this.markerCollection = value;
    }

    /**
     * Gets the value of the taxon property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.Taxon }
     *     
     */
    public Gene.Taxon getTaxon() {
        return taxon;
    }

    /**
     * Sets the value of the taxon property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.Taxon }
     *     
     */
    public void setTaxon(Gene.Taxon value) {
        this.taxon = value;
    }

    /**
     * Gets the value of the homologousAssociationCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.HomologousAssociationCollection }
     *     
     */
    public Gene.HomologousAssociationCollection getHomologousAssociationCollection() {
        return homologousAssociationCollection;
    }

    /**
     * Sets the value of the homologousAssociationCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.HomologousAssociationCollection }
     *     
     */
    public void setHomologousAssociationCollection(Gene.HomologousAssociationCollection value) {
        this.homologousAssociationCollection = value;
    }

    /**
     * Gets the value of the exonArrayReporterCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.ExonArrayReporterCollection }
     *     
     */
    public Gene.ExonArrayReporterCollection getExonArrayReporterCollection() {
        return exonArrayReporterCollection;
    }

    /**
     * Sets the value of the exonArrayReporterCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.ExonArrayReporterCollection }
     *     
     */
    public void setExonArrayReporterCollection(Gene.ExonArrayReporterCollection value) {
        this.exonArrayReporterCollection = value;
    }

    /**
     * Gets the value of the proteinCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.ProteinCollection }
     *     
     */
    public Gene.ProteinCollection getProteinCollection() {
        return proteinCollection;
    }

    /**
     * Sets the value of the proteinCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.ProteinCollection }
     *     
     */
    public void setProteinCollection(Gene.ProteinCollection value) {
        this.proteinCollection = value;
    }

    /**
     * Gets the value of the geneOntologyCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.GeneOntologyCollection }
     *     
     */
    public Gene.GeneOntologyCollection getGeneOntologyCollection() {
        return geneOntologyCollection;
    }

    /**
     * Sets the value of the geneOntologyCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.GeneOntologyCollection }
     *     
     */
    public void setGeneOntologyCollection(Gene.GeneOntologyCollection value) {
        this.geneOntologyCollection = value;
    }

    /**
     * Gets the value of the expressionArrayReporterCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.ExpressionArrayReporterCollection }
     *     
     */
    public Gene.ExpressionArrayReporterCollection getExpressionArrayReporterCollection() {
        return expressionArrayReporterCollection;
    }

    /**
     * Sets the value of the expressionArrayReporterCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.ExpressionArrayReporterCollection }
     *     
     */
    public void setExpressionArrayReporterCollection(Gene.ExpressionArrayReporterCollection value) {
        this.expressionArrayReporterCollection = value;
    }

    /**
     * Gets the value of the geneRelativeLocationCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.GeneRelativeLocationCollection }
     *     
     */
    public Gene.GeneRelativeLocationCollection getGeneRelativeLocationCollection() {
        return geneRelativeLocationCollection;
    }

    /**
     * Sets the value of the geneRelativeLocationCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.GeneRelativeLocationCollection }
     *     
     */
    public void setGeneRelativeLocationCollection(Gene.GeneRelativeLocationCollection value) {
        this.geneRelativeLocationCollection = value;
    }

    /**
     * Gets the value of the geneAliasCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.GeneAliasCollection }
     *     
     */
    public Gene.GeneAliasCollection getGeneAliasCollection() {
        return geneAliasCollection;
    }

    /**
     * Sets the value of the geneAliasCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.GeneAliasCollection }
     *     
     */
    public void setGeneAliasCollection(Gene.GeneAliasCollection value) {
        this.geneAliasCollection = value;
    }

    /**
     * Gets the value of the organOntologyCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.OrganOntologyCollection }
     *     
     */
    public Gene.OrganOntologyCollection getOrganOntologyCollection() {
        return organOntologyCollection;
    }

    /**
     * Sets the value of the organOntologyCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.OrganOntologyCollection }
     *     
     */
    public void setOrganOntologyCollection(Gene.OrganOntologyCollection value) {
        this.organOntologyCollection = value;
    }

    /**
     * Gets the value of the geneFunctionAssociationCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.GeneFunctionAssociationCollection }
     *     
     */
    public Gene.GeneFunctionAssociationCollection getGeneFunctionAssociationCollection() {
        return geneFunctionAssociationCollection;
    }

    /**
     * Sets the value of the geneFunctionAssociationCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.GeneFunctionAssociationCollection }
     *     
     */
    public void setGeneFunctionAssociationCollection(Gene.GeneFunctionAssociationCollection value) {
        this.geneFunctionAssociationCollection = value;
    }

    /**
     * Gets the value of the nucleicAcidSequenceCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.NucleicAcidSequenceCollection }
     *     
     */
    public Gene.NucleicAcidSequenceCollection getNucleicAcidSequenceCollection() {
        return nucleicAcidSequenceCollection;
    }

    /**
     * Sets the value of the nucleicAcidSequenceCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.NucleicAcidSequenceCollection }
     *     
     */
    public void setNucleicAcidSequenceCollection(Gene.NucleicAcidSequenceCollection value) {
        this.nucleicAcidSequenceCollection = value;
    }

    /**
     * Gets the value of the chromosome property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.Chromosome }
     *     
     */
    public Gene.Chromosome getChromosome() {
        return chromosome;
    }

    /**
     * Sets the value of the chromosome property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.Chromosome }
     *     
     */
    public void setChromosome(Gene.Chromosome value) {
        this.chromosome = value;
    }

    /**
     * Gets the value of the histopathologyCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.HistopathologyCollection }
     *     
     */
    public Gene.HistopathologyCollection getHistopathologyCollection() {
        return histopathologyCollection;
    }

    /**
     * Sets the value of the histopathologyCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.HistopathologyCollection }
     *     
     */
    public void setHistopathologyCollection(Gene.HistopathologyCollection value) {
        this.histopathologyCollection = value;
    }

    /**
     * Gets the value of the transcriptCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.TranscriptCollection }
     *     
     */
    public Gene.TranscriptCollection getTranscriptCollection() {
        return transcriptCollection;
    }

    /**
     * Sets the value of the transcriptCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.TranscriptCollection }
     *     
     */
    public void setTranscriptCollection(Gene.TranscriptCollection value) {
        this.transcriptCollection = value;
    }

    /**
     * Gets the value of the databaseCrossReferenceCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.DatabaseCrossReferenceCollection }
     *     
     */
    public Gene.DatabaseCrossReferenceCollection getDatabaseCrossReferenceCollection() {
        return databaseCrossReferenceCollection;
    }

    /**
     * Sets the value of the databaseCrossReferenceCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.DatabaseCrossReferenceCollection }
     *     
     */
    public void setDatabaseCrossReferenceCollection(Gene.DatabaseCrossReferenceCollection value) {
        this.databaseCrossReferenceCollection = value;
    }

    /**
     * Gets the value of the physicalLocationCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.PhysicalLocationCollection }
     *     
     */
    public Gene.PhysicalLocationCollection getPhysicalLocationCollection() {
        return physicalLocationCollection;
    }

    /**
     * Sets the value of the physicalLocationCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.PhysicalLocationCollection }
     *     
     */
    public void setPhysicalLocationCollection(Gene.PhysicalLocationCollection value) {
        this.physicalLocationCollection = value;
    }

    /**
     * Gets the value of the libraryCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.LibraryCollection }
     *     
     */
    public Gene.LibraryCollection getLibraryCollection() {
        return libraryCollection;
    }

    /**
     * Sets the value of the libraryCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.LibraryCollection }
     *     
     */
    public void setLibraryCollection(Gene.LibraryCollection value) {
        this.libraryCollection = value;
    }

    /**
     * Gets the value of the targetCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.TargetCollection }
     *     
     */
    public Gene.TargetCollection getTargetCollection() {
        return targetCollection;
    }

    /**
     * Sets the value of the targetCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.TargetCollection }
     *     
     */
    public void setTargetCollection(Gene.TargetCollection value) {
        this.targetCollection = value;
    }

    /**
     * Gets the value of the pathwayCollection property.
     * 
     * @return
     *     possible object is
     *     {@link Gene.PathwayCollection }
     *     
     */
    public Gene.PathwayCollection getPathwayCollection() {
        return pathwayCollection;
    }

    /**
     * Sets the value of the pathwayCollection property.
     * 
     * @param value
     *     allowed object is
     *     {@link Gene.PathwayCollection }
     *     
     */
    public void setPathwayCollection(Gene.PathwayCollection value) {
        this.pathwayCollection = value;
    }

    /**
     * Gets the value of the bigid property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getBigid() {
        return bigid;
    }

    /**
     * Sets the value of the bigid property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setBigid(String value) {
        this.bigid = value;
    }

    /**
     * Gets the value of the clusterId property.
     * 
     * @return
     *     possible object is
     *     {@link Long }
     *     
     */
    public Long getClusterId() {
        return clusterId;
    }

    /**
     * Sets the value of the clusterId property.
     * 
     * @param value
     *     allowed object is
     *     {@link Long }
     *     
     */
    public void setClusterId(Long value) {
        this.clusterId = value;
    }

    /**
     * Gets the value of the fullName property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getFullName() {
        return fullName;
    }

    /**
     * Sets the value of the fullName property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setFullName(String value) {
        this.fullName = value;
    }

    /**
     * Gets the value of the hugoSymbol property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getHugoSymbol() {
        return hugoSymbol;
    }

    /**
     * Sets the value of the hugoSymbol property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setHugoSymbol(String value) {
        this.hugoSymbol = value;
    }

    /**
     * Gets the value of the id property.
     * 
     * @return
     *     possible object is
     *     {@link Long }
     *     
     */
    public Long getId() {
        return id;
    }

    /**
     * Sets the value of the id property.
     * 
     * @param value
     *     allowed object is
     *     {@link Long }
     *     
     */
    public void setId(Long value) {
        this.id = value;
    }

    /**
     * Gets the value of the symbol property.
     * 
     * @return
     *     possible object is
     *     {@link String }
     *     
     */
    public String getSymbol() {
        return symbol;
    }

    /**
     * Sets the value of the symbol property.
     * 
     * @param value
     *     allowed object is
     *     {@link String }
     *     
     */
    public void setSymbol(String value) {
        this.symbol = value;
    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Chromosome" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "chromosome"
    })
    public static class Chromosome {

        @XmlElement(name = "Chromosome")
        protected gov.nih.nlm.cabio.jaxb.generated.pathways.Chromosome chromosome;

        /**
         * Gets the value of the chromosome property.
         * 
         * @return
         *     possible object is
         *     {@link gov.nih.nlm.cabio.jaxb.generated.pathways.Chromosome }
         *     
         */
        public gov.nih.nlm.cabio.jaxb.generated.pathways.Chromosome getChromosome() {
            return chromosome;
        }

        /**
         * Sets the value of the chromosome property.
         * 
         * @param value
         *     allowed object is
         *     {@link gov.nih.nlm.cabio.jaxb.generated.pathways.Chromosome }
         *     
         */
        public void setChromosome(gov.nih.nlm.cabio.jaxb.generated.pathways.Chromosome value) {
            this.chromosome = value;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneCytogeneticLocation" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "geneCytogeneticLocation"
    })
    public static class CytogeneticLocationCollection {

        @XmlElement(name = "GeneCytogeneticLocation")
        protected List<GeneCytogeneticLocation> geneCytogeneticLocation;

        /**
         * Gets the value of the geneCytogeneticLocation property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the geneCytogeneticLocation property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGeneCytogeneticLocation().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GeneCytogeneticLocation }
         * 
         * 
         */
        public List<GeneCytogeneticLocation> getGeneCytogeneticLocation() {
            if (geneCytogeneticLocation == null) {
                geneCytogeneticLocation = new ArrayList<GeneCytogeneticLocation>();
            }
            return this.geneCytogeneticLocation;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.common.domain}DatabaseCrossReference" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "databaseCrossReference"
    })
    public static class DatabaseCrossReferenceCollection {

        @XmlElement(name = "DatabaseCrossReference", namespace = "gme://caCORE.caBIO/4.3/gov.nih.nci.common.domain")
        protected List<DatabaseCrossReference> databaseCrossReference;

        /**
         * Gets the value of the databaseCrossReference property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the databaseCrossReference property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getDatabaseCrossReference().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link DatabaseCrossReference }
         * 
         * 
         */
        public List<DatabaseCrossReference> getDatabaseCrossReference() {
            if (databaseCrossReference == null) {
                databaseCrossReference = new ArrayList<DatabaseCrossReference>();
            }
            return this.databaseCrossReference;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}ExonArrayReporter" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "exonArrayReporter"
    })
    public static class ExonArrayReporterCollection {

        @XmlElement(name = "ExonArrayReporter")
        protected List<ExonArrayReporter> exonArrayReporter;

        /**
         * Gets the value of the exonArrayReporter property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the exonArrayReporter property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getExonArrayReporter().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link ExonArrayReporter }
         * 
         * 
         */
        public List<ExonArrayReporter> getExonArrayReporter() {
            if (exonArrayReporter == null) {
                exonArrayReporter = new ArrayList<ExonArrayReporter>();
            }
            return this.exonArrayReporter;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}ExpressionArrayReporter" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "expressionArrayReporter"
    })
    public static class ExpressionArrayReporterCollection {

        @XmlElement(name = "ExpressionArrayReporter")
        protected List<ExpressionArrayReporter> expressionArrayReporter;

        /**
         * Gets the value of the expressionArrayReporter property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the expressionArrayReporter property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getExpressionArrayReporter().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link ExpressionArrayReporter }
         * 
         * 
         */
        public List<ExpressionArrayReporter> getExpressionArrayReporter() {
            if (expressionArrayReporter == null) {
                expressionArrayReporter = new ArrayList<ExpressionArrayReporter>();
            }
            return this.expressionArrayReporter;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneAlias" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "geneAlias"
    })
    public static class GeneAliasCollection {

        @XmlElement(name = "GeneAlias")
        protected List<GeneAlias> geneAlias;

        /**
         * Gets the value of the geneAlias property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the geneAlias property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGeneAlias().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GeneAlias }
         * 
         * 
         */
        public List<GeneAlias> getGeneAlias() {
            if (geneAlias == null) {
                geneAlias = new ArrayList<GeneAlias>();
            }
            return this.geneAlias;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneFunctionAssociation" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "geneFunctionAssociation"
    })
    public static class GeneFunctionAssociationCollection {

        @XmlElement(name = "GeneFunctionAssociation")
        protected List<GeneFunctionAssociation> geneFunctionAssociation;

        /**
         * Gets the value of the geneFunctionAssociation property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the geneFunctionAssociation property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGeneFunctionAssociation().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GeneFunctionAssociation }
         * 
         * 
         */
        public List<GeneFunctionAssociation> getGeneFunctionAssociation() {
            if (geneFunctionAssociation == null) {
                geneFunctionAssociation = new ArrayList<GeneFunctionAssociation>();
            }
            return this.geneFunctionAssociation;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneOntology" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "geneOntology"
    })
    public static class GeneOntologyCollection {

        @XmlElement(name = "GeneOntology")
        protected List<GeneOntology> geneOntology;

        /**
         * Gets the value of the geneOntology property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the geneOntology property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGeneOntology().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GeneOntology }
         * 
         * 
         */
        public List<GeneOntology> getGeneOntology() {
            if (geneOntology == null) {
                geneOntology = new ArrayList<GeneOntology>();
            }
            return this.geneOntology;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GeneRelativeLocation" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "geneRelativeLocation"
    })
    public static class GeneRelativeLocationCollection {

        @XmlElement(name = "GeneRelativeLocation")
        protected List<GeneRelativeLocation> geneRelativeLocation;

        /**
         * Gets the value of the geneRelativeLocation property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the geneRelativeLocation property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGeneRelativeLocation().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GeneRelativeLocation }
         * 
         * 
         */
        public List<GeneRelativeLocation> getGeneRelativeLocation() {
            if (geneRelativeLocation == null) {
                geneRelativeLocation = new ArrayList<GeneRelativeLocation>();
            }
            return this.geneRelativeLocation;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Histopathology" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "histopathology"
    })
    public static class HistopathologyCollection {

        @XmlElement(name = "Histopathology")
        protected List<Histopathology> histopathology;

        /**
         * Gets the value of the histopathology property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the histopathology property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getHistopathology().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Histopathology }
         * 
         * 
         */
        public List<Histopathology> getHistopathology() {
            if (histopathology == null) {
                histopathology = new ArrayList<Histopathology>();
            }
            return this.histopathology;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}HomologousAssociation" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "homologousAssociation"
    })
    public static class HomologousAssociationCollection {

        @XmlElement(name = "HomologousAssociation")
        protected List<HomologousAssociation> homologousAssociation;

        /**
         * Gets the value of the homologousAssociation property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the homologousAssociation property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getHomologousAssociation().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link HomologousAssociation }
         * 
         * 
         */
        public List<HomologousAssociation> getHomologousAssociation() {
            if (homologousAssociation == null) {
                homologousAssociation = new ArrayList<HomologousAssociation>();
            }
            return this.homologousAssociation;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Library" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "library"
    })
    public static class LibraryCollection {

        @XmlElement(name = "Library")
        protected List<Library> library;

        /**
         * Gets the value of the library property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the library property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getLibrary().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Library }
         * 
         * 
         */
        public List<Library> getLibrary() {
            if (library == null) {
                library = new ArrayList<Library>();
            }
            return this.library;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Marker" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "marker"
    })
    public static class MarkerCollection {

        @XmlElement(name = "Marker")
        protected List<Marker> marker;

        /**
         * Gets the value of the marker property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the marker property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getMarker().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Marker }
         * 
         * 
         */
        public List<Marker> getMarker() {
            if (marker == null) {
                marker = new ArrayList<Marker>();
            }
            return this.marker;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}NucleicAcidSequence" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "nucleicAcidSequence"
    })
    public static class NucleicAcidSequenceCollection {

        @XmlElement(name = "NucleicAcidSequence")
        protected List<NucleicAcidSequence> nucleicAcidSequence;

        /**
         * Gets the value of the nucleicAcidSequence property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the nucleicAcidSequence property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getNucleicAcidSequence().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link NucleicAcidSequence }
         * 
         * 
         */
        public List<NucleicAcidSequence> getNucleicAcidSequence() {
            if (nucleicAcidSequence == null) {
                nucleicAcidSequence = new ArrayList<NucleicAcidSequence>();
            }
            return this.nucleicAcidSequence;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}OrganOntology" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "organOntology"
    })
    public static class OrganOntologyCollection {

        @XmlElement(name = "OrganOntology")
        protected List<OrganOntology> organOntology;

        /**
         * Gets the value of the organOntology property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the organOntology property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getOrganOntology().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link OrganOntology }
         * 
         * 
         */
        public List<OrganOntology> getOrganOntology() {
            if (organOntology == null) {
                organOntology = new ArrayList<OrganOntology>();
            }
            return this.organOntology;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Pathway" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "pathway"
    })
    public static class PathwayCollection {

        @XmlElement(name = "Pathway")
        protected List<Pathway> pathway;

        /**
         * Gets the value of the pathway property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the pathway property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getPathway().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Pathway }
         * 
         * 
         */
        public List<Pathway> getPathway() {
            if (pathway == null) {
                pathway = new ArrayList<Pathway>();
            }
            return this.pathway;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}GenePhysicalLocation" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "genePhysicalLocation"
    })
    public static class PhysicalLocationCollection {

        @XmlElement(name = "GenePhysicalLocation")
        protected List<GenePhysicalLocation> genePhysicalLocation;

        /**
         * Gets the value of the genePhysicalLocation property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the genePhysicalLocation property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getGenePhysicalLocation().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link GenePhysicalLocation }
         * 
         * 
         */
        public List<GenePhysicalLocation> getGenePhysicalLocation() {
            if (genePhysicalLocation == null) {
                genePhysicalLocation = new ArrayList<GenePhysicalLocation>();
            }
            return this.genePhysicalLocation;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Protein" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "protein"
    })
    public static class ProteinCollection {

        @XmlElement(name = "Protein")
        protected List<Protein> protein;

        /**
         * Gets the value of the protein property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the protein property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getProtein().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Protein }
         * 
         * 
         */
        public List<Protein> getProtein() {
            if (protein == null) {
                protein = new ArrayList<Protein>();
            }
            return this.protein;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Target" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "target"
    })
    public static class TargetCollection {

        @XmlElement(name = "Target")
        protected List<Target> target;

        /**
         * Gets the value of the target property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the target property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getTarget().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Target }
         * 
         * 
         */
        public List<Target> getTarget() {
            if (target == null) {
                target = new ArrayList<Target>();
            }
            return this.target;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Taxon" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "taxon"
    })
    public static class Taxon {

        @XmlElement(name = "Taxon")
        protected gov.nih.nlm.cabio.jaxb.generated.pathways.Taxon taxon;

        /**
         * Gets the value of the taxon property.
         * 
         * @return
         *     possible object is
         *     {@link gov.nih.nlm.cabio.jaxb.generated.pathways.Taxon }
         *     
         */
        public gov.nih.nlm.cabio.jaxb.generated.pathways.Taxon getTaxon() {
            return taxon;
        }

        /**
         * Sets the value of the taxon property.
         * 
         * @param value
         *     allowed object is
         *     {@link gov.nih.nlm.cabio.jaxb.generated.pathways.Taxon }
         *     
         */
        public void setTaxon(gov.nih.nlm.cabio.jaxb.generated.pathways.Taxon value) {
            this.taxon = value;
        }

    }


    /**
     * <p>Java class for anonymous complex type.
     * 
     * <p>The following schema fragment specifies the expected content contained within this class.
     * 
     * <pre>
     * &lt;complexType>
     *   &lt;complexContent>
     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
     *       &lt;sequence>
     *         &lt;element ref="{gme://caCORE.caBIO/4.3/gov.nih.nci.cabio.domain}Transcript" maxOccurs="unbounded" minOccurs="0"/>
     *       &lt;/sequence>
     *     &lt;/restriction>
     *   &lt;/complexContent>
     * &lt;/complexType>
     * </pre>
     * 
     * 
     */
    @XmlAccessorType(XmlAccessType.FIELD)
    @XmlType(name = "", propOrder = {
        "transcript"
    })
    public static class TranscriptCollection {

        @XmlElement(name = "Transcript")
        protected List<Transcript> transcript;

        /**
         * Gets the value of the transcript property.
         * 
         * <p>
         * This accessor method returns a reference to the live list,
         * not a snapshot. Therefore any modification you make to the
         * returned list will be present inside the JAXB object.
         * This is why there is not a <CODE>set</CODE> method for the transcript property.
         * 
         * <p>
         * For example, to add a new item, do as follows:
         * <pre>
         *    getTranscript().add(newItem);
         * </pre>
         * 
         * 
         * <p>
         * Objects of the following type(s) are allowed in the list
         * {@link Transcript }
         * 
         * 
         */
        public List<Transcript> getTranscript() {
            if (transcript == null) {
                transcript = new ArrayList<Transcript>();
            }
            return this.transcript;
        }

    }

}
